py_ecc/bls/hash.py

```diff
@@ -30,8 +35,31 @@ def hkdf_expand(prk: Union[bytes, bytearray], info: Union[bytes, bytearray], len
         text = previous + info + bytes([i + 1])
 
         # T(i + 1) = HMAC(T(i) || info || i)
-        previous = bytearray(hmac.new(prk, text, hashlib.sha256).digest())
+        previous = bytearray(hmac.new(prk, text, _sha256).digest())
         okm.extend(previous)
 
     # Return first `length` bytes.
     return okm[:length]
+
+
+def sha256(x: bytes) -> bytes:
+    m = _sha256()
+    m.update(x)
+    return m.digest()
+
+
+def xor(a: bytes, b: bytes) -> bytes:
+    return bytes(_a ^ _b for _a, _b in zip(a, b))
+
+
+def expand_message_xmd(msg: bytes, DST: bytes, len_in_bytes: int) -> bytes:
+    ell = math.ceil(len_in_bytes / HASH_TO_FIELD_B_IN_BYTES)
+    DST_prime = len(DST).to_bytes(1, 'big') + DST
```

Comment:
I was curious about the performance characteristics of the various ways to convert a single integer value to a single byte.

The three that I evaluated were:

- `value.to_bytes(1, 'big')`
- `bytes([value])`
- `struct.pack('B', value)`

Both the `to_bytes` and the `bytes([value])` approach measured the same.  `struct.pack` however seems to be about 33% faster.

Reply:
Awesome, thanks for checking this! 👍
I'll switch over to `struct.pack('B', value)`.

Reply:
I just realized there is a much dumber solution that is around 6x faster.

```python
ALL_BYTES = tuple(bytes([i]) for i in range(256))
ord_to_byte = ALL_BYTES.index
```

This retains the property of throwing an error if the integer is out of range and changes it to be a constant speed lookup.

Reply:
It took me 6 days to come around to this proposal of yours, @pipermerriam. I was really torn between the beauty of its simplicity and its hackiness. Ultimately, I think the speed up is worth it here as the speed of this library is already a bottleneck for some projects. :)